.github/workflows/terraform.yaml

```diff
@@ -1,28 +1,46 @@
-name: Check the Terraform Module
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: Check Terraform Module
 
 on:
   workflow_call:
 
 jobs:
-  build:
-    name: Terraform Checks and Plans
+  check-terraform:
+    name: Check Terraform
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./terraform
+    outputs:
+      CHECK_STATUS: "${{ env.CHECK_STATUS }}"
+
     steps:
-      - name: Checkout the repository to the runner
+      - name: Checkout Repository
         uses: actions/checkout@v4
 
-      - name: HashiCorp - Setup Terraform
+      - name: Setup Terraform
         uses: hashicorp/setup-terraform@v3
 
-      - name: Terraform Format
-        id: fmt
-        run: terraform fmt -recursive -check
-        continue-on-error: true
-
-      - name: Terraform Init
+      - name: Initialize Terraform Module
         id: init
         run: terraform init
 
-      - name: Terraform Validate
+      - name: Validate Terraform Module
         id: validate
-        run: terraform validate -no-color
\ No newline at end of file
+        run: terraform validate -no-color
+
+      - name: Validate terraform fmt
+        run: |
+          set +e 
+          echo "CHECK_STATUS=0" >> "$GITHUB_ENV"
+          terraform fmt -recursive -check
+          FMT_STATUS="$?"
+          if [[ "$FMT_STATUS" -ne 0 ]]; then
+            echo "❌ terraform fmt failed" >> "$GITHUB_STEP_SUMMARY"
+            echo "CHECK_STATUS=1" >> "$GITHUB_ENV"
+          fi
+      - name: Process check
+        if: always()
+        run: exit "$CHECK_STATUS"
```

Comment:
Does this work? Shouldn't it depend on `Validate terraform fmt`?

Reply:
Removed this step and appended it to Validate terraform fmt.

Reply:
Did you push the changes? :)